packages/core/src/package-graph/package-graph.ts

```diff
@@ -63,10 +77,14 @@ export class PackageGraph extends Map<string, PackageGraphNode> {
 
       Object.keys(graphDependencies).forEach((depName) => {
         const depNode = this.get(depName);
+        const originalSpec = graphDependencies[depName];
+        if (depName.includes('ionic')) {
+          console.log('originalspec=%s', originalSpec);
+        }
         // Yarn decided to ignore https://github.com/npm/npm/pull/15900 and implemented "link:"
         // As they apparently have no intention of being compatible, we have to do it for them.
         // @see https://github.com/yarnpkg/yarn/issues/4212
-        let spec = graphDependencies[depName].replace(/^link:/, 'file:');
+        let spec = graphDependencies[depName].replace(/^link:/, 'file:').replace(YARN_PATCH_PROTOCOL_REG_EXP, '$2');
```

Comment:
I wonder if this will have a perf impact? Wouldn't it be better to make sure that the substring does start with the text "patch:", the previous check of `^link:` does that with the use of `^` but you're not using it in your regex, so I think it might better to do something like 
```ts
if (spec.startsWith('patch:')) {
  spec = spec.replace(YARN_PATCH_PROTOCOL_REG_EXP, '$2');
}
```

Reply:
@ghiscoding I agree it's more readable and most likely more performant that way as well, thank you for the great suggestion, I updated it exactly like that!